terraform/domains/environment_domains/main.tf

```diff
@@ -1,15 +1,15 @@
 module "domains" {
-  source              = "./vendor/modules/domains//domains/environment_domains"
-  zone                = var.zone
-  front_door_name     = var.front_door_name
-  resource_group_name = var.resource_group_name
-  domains             = var.domains
-  environment         = var.environment_short
-  host_name           = var.origin_hostname
-  null_host_header    = var.null_host_header
+  for_each              = var.hosted_zone
+  source                = "./vendor/modules/domains//domains/environment_domains"
+  zone                  = each.key
+  front_door_name       = each.value.front_door_name
+  resource_group_name   = each.value.resource_group_name
+  domains               = each.value.domains
+  environment           = each.value.environment_short
+  host_name             = each.value.origin_hostname
 }
-module "records" {
+
+module "dns_records" {
```

Comment:
I think this was only used to create the old cname records and may be removed now